accelerators/nvidia.go

```diff
@@ -129,7 +139,9 @@ func (nm *NvidiaManager) Destroy() {
 // present in the devices.list file in the given devicesCgroupPath.
 func (nm *NvidiaManager) GetCollector(devicesCgroupPath string) (AcceleratorCollector, error) {
 	nc := &NvidiaCollector{}
+	nm.RLock()
 	if !nm.nvmlInitialized || len(nm.nvidiaDevices) == 0 {
+		nm.RUnlock()
```

Comment:
do we want to defer unlock here?  What if we do not enter into this if statement?

Reply:
Updated. Thanks for noticing this! (facepalm)